source/agora/node/Ledger.d

```diff
@@ -105,19 +105,20 @@ public class Ledger
             }
         }
 
-        // restore validator set from lastest blocks
-        if (this.last_block.header.height > 0)
+        // max: there may be less than 1008 blocks stored in the ledger
+        const min_height = max(long(0), long(this.last_block.header.height)
+            - Enrollment.ValidatorCycle);
```

Comment:
```suggestion
        const min_height = max(long(0), long(this.last_block.header.height)
            - Enrollment.ValidatorCycle + 1);
```
I've checked it over and over again, and I think I need to add one more, so please check it once.

Reply:
I'll add a test-case to be sure.

Reply:
The code is very simple and I love it. I also checked the added test code.
What is the height of the block at 1009?

Reply:
The code is very simple and I love it. I also checked the added test code.
What is the result of block height 1009?

Reply:
Well if the validator set is empty at block 1008, it's also empty at block 1009.

Reply:
Okay~